src/types/simple/serialize.rs

```diff
@@ -169,64 +191,120 @@ impl From<SerSimpleType> for SimpleType {
     fn from(value: SerSimpleType) -> Self {
         match value {
             SerSimpleType::Q => SimpleType::Qubit,
-            SerSimpleType::I { width } => ClassicType::Int(width).into(),
+            SerSimpleType::I { width } => HashableType::Int(width).into(),
             SerSimpleType::F => ClassicType::F64.into(),
-            SerSimpleType::S => ClassicType::String.into(),
+            SerSimpleType::S => HashableType::String.into(),
             SerSimpleType::G {
                 resources,
                 signature,
             } => ClassicType::Graph(Box::new((*resources, *signature))).into(),
             SerSimpleType::Tuple {
                 row: inner,
-                l: true,
+                c: TypeClass::Any,
             } => Container::<SimpleType>::Tuple(Box::new(inner.map_into())).into(),
             SerSimpleType::Tuple {
                 row: inner,
-                l: false,
+                c: TypeClass::Classic,
             } => {
                 Container::<ClassicType>::Tuple(Box::new(inner.try_convert_elems().unwrap())).into()
             }
+            SerSimpleType::Tuple {
```

Comment:
Yeah, every SerSimpleType container has to appear here three ways. I think the only way round this is a macro, so I'll look into that

Reply:
yes a macro would make this much more readable

Reply:
Et, voila! (My first macro!)
I particularly like that I was worried I'd get some of the Simple/Classic/Hashable stuff the wrong way round if I had to write it that many times, but now I only have to write that logic out once :)